### PR TITLE
feat!: upgrade to Django 4.2 LTS

### PR DESCRIPTION
**Pull request by awais786**
Upgrading django42.

### CODE REVIEW

File: requirements/django.txt [hunk 1]
```diff
@@ -1 +1 @@
-django==3.2.22
+django==4.2.6
```
Comment (awais786) [inline requirements/django.txt:1]: Django version.

File: requirements/base.txt [hunk 2]
```diff
@@ -18,11 +20,11 @@ chardet==3.0.4
     # via requests
 click==8.1.7
     # via edx-django-utils
-cryptography==41.0.4
+cryptography==41.0.5
     # via pyjwt
-django==3.2.22
+django==4.2.6
```
Comment (awais786) [inline requirements/base.txt:25]: Django

**Review by UsamaSadiq — APPROVED** [review]
All changes look good. Verified that make upgrade is also working successfully through github action.

**Review by mattcarter — APPROVED** [review]
approved

**Pull request merged**